engine/src/cmd/collide2/CSopcodecollider.cpp

```diff
@@ -33,25 +33,26 @@
 #include "CSopcodecollider.h"
 #include "opcodeqsqrt.h"
 #include "opcodeqint.h"
-#include "opcodegarray.h"
+// #include "opcodegarray.h"
 #define _X 1000
 
 #undef _X
 
 using namespace Opcode;
+using namespace VegaStrike;
 
-static CS_DECLARE_GROWING_ARRAY_REF (pairs,csCollisionPair);
+static std::shared_ptr<vs_vector<csCollisionPair>> pairs = std::make_shared<vs_vector<csCollisionPair>>();
 
 csOPCODECollider::csOPCODECollider (const std::vector <mesh_polygon> &polygons)
 {
-	m_pCollisionModel = 0;
-	vertholder = 0;
-	pairs.IncRef();
-	TreeCollider.SetFirstContact(true);
-	TreeCollider.SetFullBoxBoxTest(false);
-	TreeCollider.SetTemporalCoherence(false);
-	opcMeshInt.SetCallback (&MeshCallback, this);
-	GeometryInitialize (polygons);
+    m_pCollisionModel = nullptr;
+    vertholder = nullptr;
+    //pairs.IncRef();
```

Comment:
I'd be a little caution about removing reference counting

Reply:
My understanding was that this was a custom implementation of `std::shared_ptr`, before there was such a thing. So I just used that instead. But perhaps `boost::intrusive_ptr` would be a better fit, I don't know.

Note also that the `pairs` variable is static anyway.

Reply:
I ended up using a static `vector` directly, without a `shared_ptr` around it. The code seems to be working well now.